src/utils/threads/thread_pool.cpp

```diff
@@ -0,0 +1,161 @@
+#include "lbann/utils/threads/thread_pool.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+namespace lbann {
+
+thread_pool::thread_pool()
+  : thread_joiner_{threads_},
+    all_work_done_{false}
+{
+}
+
+thread_pool::thread_pool(size_type max_threads)
+  : thread_pool()
+{
+  size_type num_threads = std::max(max_threads,size_type{1});
+  this->launch_threads(num_threads);
+}
+
+void thread_pool::launch_threads(size_type num_threads)
+{
+  threads_.reserve(num_threads);
+
+  // Try to launch each worker thread
+  try
+  {
+    for (size_type cnt = 0; cnt < num_threads; ++cnt) {
+      threads_.emplace_back(&thread_pool::do_thread_work_,this);
+      // pthread_t my_thread_native = threads_[cnt].native_handle();
```

Comment:
Is this code possibly going to be resurrected? If not (and even if so), perhaps it should be deleted.